test/UnionIterator-test.js

```diff
@@ -580,9 +547,18 @@ describe('UnionIterator', () => {
       await new Promise(resolve => iterator.on('end', resolve));
 
       sourcesIterator.closed.should.be.true;
+    });
+
+    it('should close iterators that have started to be read', async () => {
+      iterator.read();
+      iterator.close();
+
+      await new Promise(resolve => iterator.on('end', resolve));
+
+      sourcesIterator.closed.should.be.true;
 
       sources[0].closed.should.be.true;
-      sources[1].closed.should.be.true;
+      sources[1].closed.should.be.false;
```

Comment:
This is an example of where a source could not be destroyed and hence closed because it had not yet been pulled from `this._source`.